source/common/tracing/zipkin/span_buffer.cc

```diff
@@ -2,7 +2,7 @@
 
 namespace Zipkin {
 
-bool SpanBuffer::addSpan(SpanPtr&& span) {
+bool SpanBuffer::addSpan(Span&& span) {
```

Comment:
I forgot to define the Span move constructor!  It was my intention to have defined the move constructor.  Defining it will avoid the copies.

The reason I moved away from passing pointers above is as follows. Envoy deals with a unique_ptr to a span. That is enforced by the the virtual method Tracing::Driver::startSpan(), which ZipkinDriver and LightStepDriver implement.
Envoy also defines the virtual method Span::finishSpan(), which must be implemented by LightStepSpan and ZipkinSpan. This method has to perform the needed actions on the span before it can be sent to the corresponding backend (LightStep or Zipkin). One of the things that both the LightStep and the Zipkin-related code do when "finishing" a span is to buffer it so that a batch of spans can be sent together to the backend.  Thus, the finish() method of the Span class must insert "this" span into the buffer. It turns out that doing so while passing around a unique_ptr is not safe. Thus, both the LightStep and the Zipkin code do a std::move(*this) to pass the object via move semantics into the buffer (Granted, I forgot to define the Span move constructor!).  It seems to me that we can either pass the object via move semantics (as it is done today), or change the Envoy base Tracing interface to, perhaps, use a shared_ptr to a Span instead. The latter will require changes to the Lightstep-related code.